app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user, class_name: 'User', foreign_key: 'user_id'
+  has_many :comments
+  has_many :likes
+
+  def update_post_counter
+    user.update(post_counter: user.posts.count)
+  end
```

Comment:
- [OPTIONAL] It is a good practice to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private. It should look something similar to this :point_down: 

```
    # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end

```